src/codecs/tiff.rs

```diff
@@ -173,9 +173,30 @@ impl<'a, R: 'a + Read + Seek> ImageDecoder<'a> for TiffDecoder<R> {
         self.color_type
     }
 
-    fn into_reader(mut self) -> ImageResult<Self::Reader> {
+    fn set_limits(&mut self, limits: crate::io::Limits) -> ImageResult<()> {
+        limits.check_support(&crate::io::LimitSupport::default())?;
+
+        let (width, height) = self.dimensions();
+        limits.check_dimensions(width, height)?;
+
+        let max_alloc = limits.max_alloc.unwrap_or(u64::MAX);
+        let max_intermediate_alloc = max_alloc.saturating_sub(self.total_bytes());
+
+        let mut tiff_limits: tiff::decoder::Limits = Default::default();
+        tiff_limits.decoding_buffer_size =
+            usize::try_from(max_alloc - max_intermediate_alloc).unwrap_or(usize::MAX);
+        tiff_limits.intermediate_buffer_size =
+            usize::try_from(max_intermediate_alloc).unwrap_or(usize::MAX);
+        tiff_limits.ifd_value_size = tiff_limits.intermediate_buffer_size;
+        self.inner = Some(self.inner.take().unwrap().with_limits(tiff_limits));
```

Comment:
This option-wrapping is avoided if we had a simpler `set_limits`. Do you feel like it's worth it to open an issue on `tiff` upstream? I don't want to delay this PR in any case, we can revert back to a non-option field at a later stage with a bumped dependency.

Reply:
Not sure it is worth creating an issue now. The tiff crate's interface honestly could use a overhaul overall